includes/process-donation.php

```diff
@@ -286,6 +286,11 @@ function give_donation_form_validate_fields() {
 		give_set_error( 'invalid_honeypot', esc_html__( 'Honeypot field detected. Go away bad bot!', 'give' ) );
 	}
 
+	// Check spam detect.
+	if ( is_spam_donation() ) {
+		give_set_error( 'invalid_donation', __( 'Spam detected. Go away bad bot!', 'give' ) );
```

Comment:
Please update message `This donation has been flagged as spam. Please try again.`